examples/pytorch/ogb/directional_GSN/main.py

```diff
@@ -0,0 +1,433 @@
+        h = h * snorm_n
+        h = self.batchnorm_h(h)
+        h = F.relu(h)
+        if self.residual:
+            h = h_in + h
+
+        h = F.dropout(h, self.dropout, training=self.training)
+
+        return h
+
+class MLPReadout(nn.Module):
+
+    def __init__(self, input_dim, output_dim, L=2):  # L=nb_hidden_layers
+        super().__init__()
+        list_FC_layers = [nn.Linear(input_dim // 2 ** l, input_dim // 2 ** (l + 1), bias=True) for l in range(L)]
+        list_FC_layers.append(nn.Linear(input_dim // 2 ** L, output_dim, bias=True))
+        self.FC_layers = nn.ModuleList(list_FC_layers)
+        self.L = L
+
+    def forward(self, x):
+        y = x
+        for l in range(self.L):
+            y = self.FC_layers[l](y)
+            y = F.relu(y)
+        y = self.FC_layers[self.L](y)
+        return y
+
+class DGNNet(nn.Module):
+    def __init__(self, hidden_dim, out_dim, dropout, n_layers, avg_d):
+        super().__init__()
+        self.aggregators = "mean sum max dir1-dx"
+        self.avg_d = avg_d
+        self.residual = False
+
+        self.embedding_h = AtomEncoder(emb_dim=hidden_dim)
+
+        # retrieve the aggregators and scalers functions
+        self.aggregators = [AGGREGATORS[aggr] for aggr in self.aggregators.split()]
+
+        self.layers = nn.ModuleList([DGNLayer(in_dim=hidden_dim, out_dim=hidden_dim, dropout=dropout, 
+                      residual=self.residual, aggregators=self.aggregators,
+                      avg_d=self.avg_d) for _ in range(n_layers - 1)])
+        self.layers.append(DGNLayer(in_dim=hidden_dim, out_dim=out_dim, dropout=dropout,
+                                    residual=self.residual, aggregators=self.aggregators, 
+                                    avg_d=self.avg_d))
+
+        # 128 out dim since ogbg-molpcba has 128 tasks
+        self.MLP_layer = MLPReadout(out_dim, 128)
+
+    def forward(self, g, h, e, snorm_n):
+        h = self.embedding_h(h)
+
+        for i, conv in enumerate(self.layers):
+            h_t = conv(g, h, e, snorm_n)
+            h = h_t
+
+        g.ndata['h'] = h
+
+        hg = dgl.mean_nodes(g, 'h')
+
+        return self.MLP_layer(hg)
+
+    def loss(self, scores, labels):
+        is_labeled = labels == labels
+        loss = torch.nn.BCEWithLogitsLoss()(scores[is_labeled], labels[is_labeled].float().to('cuda'))
+        return loss
+
+def train_epoch(model, optimizer, device, data_loader):
+    model.train()
+    epoch_loss = 0
+    epoch_train_AP = 0
+    list_scores = []
+    list_labels = []
+    for iter, (batch_graphs, batch_labels, batch_snorm_n, batch_snorm_e) in enumerate(data_loader):
+        batch_x = batch_graphs.ndata['feat'].to(device)  # num x feat
+        batch_e = batch_graphs.edata['feat'].to(device)
+        batch_snorm_e = batch_snorm_e.to(device)
+        batch_snorm_n = batch_snorm_n.to(device)
+        batch_labels = batch_labels.to(device)
+        batch_graphs = batch_graphs.to(device)
```

Comment:
```suggestion
batch_graphs = batch_graphs.to(device)
batch_x = batch_graphs.ndata['feat']
batch_e = batch_graphs.edata['feat']
batch_snorm_n = batch_snorm_n.to(device)
batch_labels = batch_labels.to(device)
```

`batch_graphs = batch_graphs.to(device)` also moves all its features to the target device.